### DISCUSSION TITLE
FATAL ERROR: Ineffective mark-compacts near heap limit Allocation failed - JavaScript heap out of memory

### DISCUSSION THREAD

**Discussion by LGmatrix13** [vercel/next.js#26698] [Help]
I keep getting this error with npm run dev. I am pulling my hair out, there is no solution that has worked thus far. Oddly, I went away for 4 days (everything was working) and then came back to this. Is there any solution? I am using the latest version of next, node, and every other dependency.

Here is my package.json:
```
{
  "name": "prisma-next-auth",
  "version": "1.0.0",
  "description": "",
  "main": "index.js",
  "scripts": {
    "start": "next start",
    "dev": "next dev",
    "build": "next build"
  },
  "keywords": [],
  "author": "",
  "license": "MIT",
  "dependencies": {
    "@prisma/client": "^2.25.0",
    "gray-matter": "^4.0.2",
    "marked": "^2.0.1",
    "next": "^11.0.1",
    "next-auth": "^3.27.1",
    "react": "^17.0.2",
    "react-dom": "^17.0.2"
  },
  "devDependencies": {
    "@prisma/cli": "^2.7.1",
    "prisma": "^2.25.0"
  }
}
```

**Comment by devdogfish** [comment]
I had the same bug and after hours of digging found the error in my code. It was an infinite loop caused by a mistake in the JWT tokens and authentication flow, where a condition triggered the token function to repeat endlessly, eventually consuming all the memory. 

That is why increasing memory also didn't work.

The cause might be different for you, but just a tip: check your code for infinite loops - especially in the next-auth options file since you are using next-auth.

**Comment by yakshcore** [comment]
Yo, I’ve been there — took a short break, came back, and boom... `npm run dev` just refuses to vibe anymore 

In my case, the issue turned out to be something totally not obvious:
 **An accidental infinite loop in my NextAuth config** — specifically in the `jwt` or `session` callback.

What happened was:
I had a condition that kept triggering the token refresh, which then re-triggered itself endlessly. No errors in the terminal, just high memory usage and the dev server crashing or freezing. Increasing memory limits didn’t help because it was purely a logic loop.

So if you're using NextAuth (which I see you are), I’d strongly suggest:

* Double-check your `next-auth` config (`pages/api/auth/[...nextauth].js`)
* Look at any logic inside `callbacks: { jwt, session }`
* Watch out for any logic like `if (token) { return await refreshToken(token) }` without proper exit conditions

 Temporary hack: try commenting out your custom callbacks and slowly reintroducing them to isolate the cause.

Hope this saves you hours of hair-pulling like it did for me.  Let me know if you want help checking the exact file.
